packages/generators/app/src/utils/git.ts

```diff
@@ -0,0 +1,35 @@
+import execa from 'execa';
+
+async function isInGitRepository(rootDir: string) {
+  try {
+    await execa('git', ['rev-parse', '--is-inside-work-tree'], { stdio: 'ignore', cwd: rootDir });
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+async function isInMercurialRepository(rootDir: string) {
+  try {
+    await execa('hg', ['-cwd', '.', 'root'], { stdio: 'ignore', cwd: rootDir });
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+export default async function tryGitInit(rootDir: string) {
+  try {
+    await execa('git', ['--version'], { stdio: 'ignore' });
+    if ((await isInGitRepository(rootDir)) || (await isInMercurialRepository(rootDir))) {
+      return false;
+    }
+
+    await execa('git', ['init'], { stdio: 'ignore', cwd: rootDir });
+
+    return true;
+  } catch (e) {
+    console.log(e);
```

Comment:
Shouldn't we write to stderr instead?
Also, do we really want to log a random error message inherited from the shell if something fails? (e.g. "Git is not installed, skipping repository init" or something similar)

Reply:
> "Git is not installed, skipping repository init" 

For git --version, but we could have another message for git init